pkg/instance/handler.go

```diff
@@ -102,80 +116,61 @@ func (h Handler) Create(c *gin.Context) {
 		return
 	}
 
-	savedInstance, err := h.instanceService.Create(instance)
-	if err != nil {
-		_ = c.Error(err)
-		return
-	}
+	if request.SourceInstance != 0 {
+		sourceInstance, err := h.instanceService.FindWithDecryptedParametersById(request.SourceInstance)
+		if err != nil {
+			_ = c.Error(err)
+			return
+		}
 
-	c.JSON(http.StatusCreated, savedInstance)
-}
+		canWriteSource := handler.CanWriteInstance(userWithGroups, sourceInstance)
+		if !canWriteSource {
+			err := apperror.NewUnauthorized(fmt.Sprintf("write access to source instance (id: %d) denied", sourceInstance.ID))
+			_ = c.Error(err)
+			return
+		}
 
-// Restart instance
-// swagger:route POST /instances/{id}/restart restartInstance
-//
-// Restart instance
-//
-// Security:
-//  oauth2:
-//
-// responses:
-//   202:
-//   401: Error
-//   403: Error
-//   404: Error
-//   415: Error
-func (h Handler) Restart(c *gin.Context) {
-	idParam := c.Param("id")
-	id, err := strconv.ParseUint(idParam, 10, 64)
-	if err != nil {
-		_ = c.AbortWithError(http.StatusBadRequest, fmt.Errorf("failed to parse id: %s", err))
-		return
-	}
+		if sourceInstance.DeployLog == "" {
+			err := fmt.Errorf("source instance (%s) not deployed", sourceInstance.Name)
```

Comment:
For strings that are user provided I would default to using `%q` just in case there are any whitespace issues. You could argue that since we validate using `dns_rfc1035_label` that should not be a problem. I would still default to it as it does not hurt here and reduces the chance of us missing it elsewhere where it hurts.